addon/components/cfb-form-editor/general.js

```diff
@@ -0,0 +1,83 @@
+import Component from "@ember/component";
+import layout from "../../templates/components/cfb-form-editor/general";
+import { task } from "ember-concurrency";
+import gql from "graphql-tag";
+import { ComponentQueryManager } from "ember-apollo-client";
+import validations from "../../validations/form";
+import v4 from "uuid/v4";
+
+export default Component.extend(ComponentQueryManager, {
+  layout,
+
+  validations,
+
+  data: task(function*() {
+    return yield this.get("apollo").watchQuery(
+      {
+        query: gql`
+          query FormGeneral($id: ID!) {
+            node(id: $id) {
+              ... on Form {
+                id
+                name
+                slug
+                description
+              }
+            }
+          }
+        `,
+        variables: { id: btoa(`Form:${this.get("slug")}`) },
```

Comment:
Are you encoding the slug because it might not be url-safe?

Reply:
No, I'm generating the global GraphQL ID which consists the object type and the slug concatenated and base64 encoded. This is part of the Relay specification.